modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
-            } else {
-                proxy.addExposedPort(port.getOriginalPort());
-            }
-        }
-
-        proxy.setWaitStrategy(null);
-        proxy.start();
-
-        ExecCreateCmdResponse createCmdResponse = dockerClient
-            .execCreateCmd(proxy.getContainerId())
-            .withCmd(
-                "sh",
-                "-c",
-                Stream.of(CouchbasePort.values())
-                    .map(port -> {
-                        return "/usr/bin/socat " +
-                            "TCP-LISTEN:" + port.getOriginalPort() + ",fork,reuseaddr " +
-                            "TCP:" + networkAlias + ":" + getMappedPort(port);
-                    })
-                    .collect(Collectors.joining(" & ", "true", ""))
-            )
-            .exec();
-
-        dockerClient.execStartCmd(createCmdResponse.getId())
-            .start()
-            .awaitCompletion(10, TimeUnit.SECONDS);
+    public final String getUsername() {
+        return username;
     }
 
-    @Override
-    public List<Integer> getExposedPorts() {
-        return proxy.getExposedPorts();
+    public final String getPassword() {
+        return password;
     }
 
-    @Override
-    public String getContainerIpAddress() {
-        return proxy.getContainerIpAddress();
+    public String getConnectionString() {
+        return String.format("couchbase://%s:%d", getContainerIpAddress(), getMappedPort(KV_PORT));
     }
 
     @Override
-    public Integer getMappedPort(int originalPort) {
-        return proxy.getMappedPort(originalPort);
-    }
+    protected void configure() {
+        super.configure();
+
+        WaitAllStrategy waitStrategy = new WaitAllStrategy();
+
+        // Makes sure that all nodes in the cluster are healthy.
+        waitStrategy = waitStrategy.withStrategy(
+            new HttpWaitStrategy()
+                .forPath("/pools/default")
+                .forPort(MGMT_PORT)
+                .withBasicCredentials(username, password)
+                .forStatusCode(200)
+                .forResponsePredicate(response -> {
+                    try {
+                        return Optional.of(MAPPER.readTree(response))
+                            .map(n -> n.at("/nodes/0/status"))
+                            .map(JsonNode::asText)
+                            .map("healthy"::equals)
+                            .orElse(false);
+                    } catch (IOException e) {
+                        logger().error("Unable to parse response {}", response);
+                        return false;
+                    }
+                })
+        );
+
+        if (enabledServices.contains(CouchbaseService.QUERY)) {
+            waitStrategy = waitStrategy.withStrategy(
+                new HttpWaitStrategy()
+                    .forPath("/admin/ping")
+                    .forPort(QUERY_PORT)
+                    .withBasicCredentials(username, password)
+                    .forStatusCode(200)
```

Comment:
see above

Reply:
Let's not :D
Really does not help, especially when someone who knows Couchbase wants to see which endpoint is used to check the liveness here

Reply:
Why does knowing about status code 200 help?